File: java/v1/src/main/java/com/api/v1/dtos/customer/CustomerRegistrationRequestDto.java
package com.api.v1.dtos.customer;

import java.time.LocalDate;
import com.api.v1.annotations.customer.SSN;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

public record CustomerRegistrationRequestDto(
        @NotBlank String firstName,
        String middleName,
        @NotBlank String lastName,
        @SSN String ssn,
        @NotNull LocalDate birthDate,
        @NotBlank @Email String email,
        @NotBlank String address,
        @NotBlank @Size(min=10, max=10) String phoneNumber
) {
}

File: v2/src/main/kotlin/com/api/v2/customer/dtos/CustomerRegistrationRequestDto.kt
package com.api.v2.customer.dtos

import java.time.LocalDate

data class CustomerRegistrationRequestDto(
    var firstName: String,
    var middleName: String,
    var lastName: String,
    var ssn: String,
    var birthDate: LocalDate,
    var email: String,
    var phoneNumber: String,
    var gender: String,
)